--- a/custom.menu.php	2017-09-22 14:48:47.726716529 +0200
+++ b/custom.menu.php	2017-09-22 14:53:23.400056545 +0200
@@ -38,8 +38,16 @@
 // INSERT YOUR CUSTOM EXTERNAL or INTERNAL LINKS HERE! //
 $quickboxURL = "https://quickbox.io/donate/";
 
+$TransmissionWebControl = "http://" . $_SERVER['HTTP_HOST'] . ":9091/transmission/web/";
+//OtakuNekoP add Transmission Support 20170225
+
 ?> <!-- Do not remove this closing (?>) -->
 
 
 <!-- // INSERT YOUR CUSTOM EXTERNAL or INTERNAL MENU ITEMS HERE! // -->
+
+<li><a href="<?php echo "$TransmissionWebControl"; ?>" class="grayscale" target="_blank"><img src="img/brands/transmission.png" class="brand-ico"> <span><?php echo T('Transmission Web Control'); ?></span></a></li>
+<li><a href="/<?php echo "$username"; ?>.transmission.downloads" class="grayscale" target="_blank"><img src="img/brands/transmission.png" class="brand-ico"> <span><?php echo T('Transmission Downloads'); ?></span></a></li>
+<!-- OtakuNekoP add Transmission Support 20170225 -->
+
 <li><a href="<?php echo "$quickboxURL"; ?>" class="grayscale" target="_blank"><i class="fa fa-heart text-danger"></i> <span><?php echo T('DONATE'); ?></span></a></li>
